### ISSUE TITLE
Parsing error: Unexpected token )

### ISSUE THREAD

**Issue by denenr** [eslint/eslint#11024]
**Tell us about your environment**

* **ESLint Version:** v5.7.0
* **Node Version:** v8.4.0
* **npm Version:** 6.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
env:
  browser: true
extends: 'eslint:recommended'
parserOptions:
  ecmaVersion: 5
rules:
  indent:
    - error
    - 4
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - error
    - single
  semi:
    - error
    - always
  eqeqeq:
    - error
    - always
  comma-dangle:
    - error
    - arrays: always-multiline
      objects: always-multiline
      imports: always-multiline
      exports: always-multiline
      functions: always-multiline

  no-alert: error
  no-eval: error
  no-implied-eval: error
  no-extend-native: error
  block-scoped-var: error
  wrap-iife: error
  eol-last: error

  complexity:
    - warn
    - max: 8
  no-warning-comments:
    - warn
    - terms: ["todo", "fix"]
      location: "start"
  max-len:
    - warn
    - 80

  no-extra-parens: warn
  keyword-spacing: warn
  no-trailing-spaces: warn
  spaced-comment: warn
  radix: warn

  brace-style:
    - warn
    - 1tbs
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function CanvasImage() {
    // -snip
}

CanvasImage.prototype.draw = function (context) {
    context.drawImage(
        this.image,
        this.startX,
        this.startY,
        this.endX,
        this.endY,
    );
};
```

```bash
eslint canvasImage.js
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
At first I had this:

```
/Users/dennis/Projects/Apps/Journal/canvasImage.js
  11:18  error  Missing trailing comma  comma-dangle

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

So I went ahead and added the comma, but then I got this:

```
/Users/dennis/Projects/Apps/Journal/canvasImage.js
  12:5  error  Parsing error: Unexpected token )

✖ 1 problem (1 error, 0 warnings)
```

The parser seems unable to handle a comma after the last parameter to a function, but there is an option to the linter for requiring a comma.

**Are you willing to submit a pull request to fix this bug?**
At the moment I don't even know where to start, but if someone could point me in the right direction I'd definitely be willing to help!

**Comment by Aladdin-ADD** [comment]
2 ways to solve it:
+ if you don't want to enforce trailing-function-commas, you can disable the option `functions: never
`, then the first error `"Missing trailing comma  comma-dangle"` should gone.
```yml
  comma-dangle:
    - error
    - arrays: always-multiline
      objects: always-multiline
      imports: always-multiline
      exports: always-multiline
      functions: never
```
+ if you want to enforce trailing-function-commas(it was introduced in es2017), you need specify "ecmaVersion: 2017"

**Comment by Aladdin-ADD** [comment]
you might want https://eslint.org/docs/rules/comma-dangle#functions and https://eslint.org/docs/user-guide/configuring#specifying-parser-options 😄

**Comment by kaicataldo** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
